mdtraj/core/topology.py

```diff
@@ -1144,7 +1219,7 @@ def select_expression(self, selection_string):
         fmt_string = "[atom.index for atom in topology.atoms if {condition}]"
         return fmt_string.format(condition=condition)
 
-    def select(self, selection_string):
+    def select(self, selection_string: str) -> NDArray[np.int32]:
```

Comment:
Is `np.int32` right here? This function creates a NumPy array on line 1235

```
indices = np.array([a.index for a in self.atoms if filter_func(a)])
```

I would actually guess it would be `np.int64` most of the time. It may be safest here to just type hint it as `NDArray[np.integer]`